### PR TITLE
feat: add group mentions

### PR DESCRIPTION
**Pull request by salihudickson**
## Description
This PR adds the ability for users to mention entire groups in a document. This will also notify users in the group that they have been mentioned in a document.

fixes: #10307 

### Mentioning:

https://github.com/user-attachments/assets/7bf10631-7131-4c1a-bc8b-6690d6f6662c

### Notification:

https://github.com/user-attachments/assets/b4960e8d-3643-47da-88fe-8291622b9b3d

### CODE REVIEW

**Comment by salihudickson** [comment]
hey @tommoor, I still have a bit of work to do on this, but could you take a look and let me know what you think of my approach.

**Comment by tommoor** [comment]
Approach looks solid. It's probably worth a new notification type to distinguish between a personal and a group mention

**Comment by salihudickson** [comment]
Thanks.

I also wanted to ask, is there a particular reason that there isn't consistent component or e2e testing across the client-side. Or is it just something you haven't gotten to yet?

**Comment by tommoor** [comment]
Mostly I don't care for it, more hassle than it's worth. Open to suggestions though

**Comment by tommoor** [comment]
Note https://github.com/outline/outline/pull/10318 just merged, so will affect use of `v4()` in this PR

**Comment by salihudickson** [comment]
> Mostly I don't care for it, more hassle than it's worth. Open to suggestions though.

🤔 often i find that the biggest benefit of these kinds of tests is really just making sure we don't accidentally break something important. Since you've gotten this far without them, if you don't feel they're necessary, I'm inclined to agree.

That said, would you be up for opening a discussion about adding some high-level E2E tests for only top-level core functionality? This would let us zero in on areas where the benefit clearly outweighs the maintenance cost. We wouldn't be trying to hit full test coverage, just making sure we don't have to manually check that core features are still working after multi-module changes or refactoring.

**Comment by salihudickson** [comment]
hey @tommoor, I think this is ready for another round of reviews.

File: server/queues/tasks/CommentCreatedNotificationsTask.ts [hunk 1]
```diff
@@ -77,6 +83,58 @@ export default class CommentCreatedNotificationsTask extends BaseTask<CommentEve
       }
     }
 
+    // send notifications to users in mentioned groups
```
Comment (tommoor) [inline server/queues/tasks/CommentCreatedNotificationsTask.ts:86]: Approving for v1 but I feel like we should split this into it's own task for resiliency, especially as groups can have many members.

**Review by tommoor — APPROVED** [review]
